### PR TITLE
Use and enforce expression body for properties and indexers

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Tools/ReleaseNotesUtil/RuleInfo.cs
Comment (sharwell) [inline src/Tools/ReleaseNotesUtil/RuleInfo.cs:64]: ⬆️ This change is the reason I do not like this rule.

**Review by sharwell — COMMENTED** [review]
Not going to block this if someone else wants it, but not a rule I would turn on for my own repositories in its current form. 😄

File: src/Tools/ReleaseNotesUtil/RuleInfo.cs
Comment (Evangelink) [reply src/Tools/ReleaseNotesUtil/RuleInfo.cs:64]: Is there any ticket on Roslyn to enable some tweak of the rule? We could think of some option that would not suggest expression-body if the content is a ternary for example or if it is a "complex" expression (to be defined).

**Comment by Evangelink** [comment]
@sharwell I am fine if we close this one out. I am just trying to apply some common styles/pattern across the repro, it would be easier if we could have some chat from time to time on the directions we(you) want to take.

File: src/Utilities/Compiler/CodeMetrics/ComputationalComplexityMetrics.cs
Comment (Youssef1313) [inline src/Utilities/Compiler/CodeMetrics/ComputationalComplexityMetrics.cs:418]: There were 2 comments, one of them was removed.
Luckily, I feel the one that was removed is really extra and unneeded, but the codefix should preserve trivia correctly.

Comment (Evangelink) [reply src/Utilities/Compiler/CodeMetrics/ComputationalComplexityMetrics.cs:418]: I did the removal manually because I wasn't sure how to merge the 2 comments. I have also created an issue on roslyn because the code-fix breaks trivia.s

**Comment by Evangelink** [comment]
@mavasani Let me know if you feel like we should use expression-bodies or drop this PR.

**Comment by mavasani** [comment]
> Let me know if you feel like we should use expression-bodies or drop this PR.

These analyzers are supported on build in the CodeStyle NuGet and the .NET 5.0 SDK, so I think your PR is good dogfooding exercise for enforcing them on build. Let's go ahead with the PR.

File: .editorconfig
Comment (mavasani) [inline .editorconfig:99]: All style rules are defaulted to warnings, so this is redundant.
```suggestion
```

https://github.com/dotnet/roslyn-analyzers/blob/5102a0d89ef178b0ca741634ee006597bf5df2ed/.editorconfig#L188-L189

File: .editorconfig
Comment (mavasani) [inline .editorconfig:96]:
```suggestion
```

File: src/Tools/ReleaseNotesUtil/RuleInfo.cs
Comment (mavasani) [reply src/Tools/ReleaseNotesUtil/RuleInfo.cs:64]: I don't understand how the original code is more readable then the new code though. I think we need to do either of 2 things here:

1. Rewrite the code in block body form breaking down into multiple statements.
2. Use the expression body form suggested in this PR.

I am fine with either approach. I think the form in the original code is probably the least preferred form.

File: src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DictionaryAnalysisData.cs
Comment (mavasani) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DictionaryAnalysisData.cs:60]: I think we may want to consider loosening the analyzer here and not report an issue when there are comments on a separate line in the body. I don't think users would prefer expression body with comments in between the `=>` and the expression. @Evangelink can you file a Roslyn issue for this?

**Review by mavasani — APPROVED** [review]
Overall LGTM. Lets try this out for a bit, and if the analyzer gets too annoying we can downgrade it to an IDE suggestion in future.

@Evangelink I'll wait for you to respond to comments before merge.

File: src/Utilities/Compiler/CodeMetrics/ComputationalComplexityMetrics.cs
Comment (Evangelink) [reply src/Utilities/Compiler/CodeMetrics/ComputationalComplexityMetrics.cs:418]: See https://github.com/dotnet/roslyn/issues/50181

File: src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DictionaryAnalysisData.cs
Comment (Evangelink) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DictionaryAnalysisData.cs:60]: Ticket created https://github.com/dotnet/roslyn/issues/50229

File: src/Tools/ReleaseNotesUtil/RuleInfo.cs
Comment (Evangelink) [reply src/Tools/ReleaseNotesUtil/RuleInfo.cs:64]: Let me know your preferences.

**Review by mavasani — APPROVED** [review]

**Comment by Evangelink** [comment]
> @Evangelink I'll wait for you to respond to comments before merge.

I think I answered them.

File: src/Tools/ReleaseNotesUtil/RuleInfo.cs
Comment (CyrusNajmabadi) [reply src/Tools/ReleaseNotesUtil/RuleInfo.cs:64]: We have a value of `when_on_single_line` for this feature.

**Pull request merged**
